File: src/main/java/com/soze/r2d/R2DNode.java
package com.soze.r2d;

import com.badlogic.gdx.scenes.scene2d.Actor;
import com.badlogic.gdx.scenes.scene2d.Group;

import java.util.ArrayList;
import java.util.List;

public class R2DNode {

  private Object type = "VOID";
  private Component component = null;
  private Actor actor = null;
  private Group group = null;
  private R2DNode parent = null;
  private List<R2DNode> children = new ArrayList<>();

  public R2DNode() {

  }

  public Object getType() {
    return type;
  }

  public void setType(Object type) {
    this.type = type;
  }

  public Component getComponent() {
    return component;
  }

  public void setComponent(Component component) {
    this.component = component;
  }

  public Actor getActor() {
    return actor;
  }

  public void setActor(Actor actor) {
    this.actor = actor;
  }

  public Group getGroup() {
    return group;
  }

  public void setGroup(Group group) {
    this.group = group;
  }

  public R2DNode getParent() {
    return parent;
  }

  public void setParent(R2DNode parent) {
    this.parent = parent;
  }

  public List<R2DNode> getChildren() {
    return children;
  }

  public void setChildren(List<R2DNode> children) {
    this.children = children;
  }

  @Override
  public String toString() {
    return "R2DNode{" +
             "\ntype=" + type +
             "\ncomponent=" + component +
             "\nactor=" + actor +
             "\ngroup=" + group +
             "\nparent=" + parent +
             "\nchildren=" + children +
             '}';
  }
}

File: src/main/kotlin/com/soze/r2d/R2DNode.kt
package com.soze.lifegame.uibuilder

import com.badlogic.gdx.scenes.scene2d.Actor
import com.badlogic.gdx.scenes.scene2d.Group

data class R2DNode(
	var type: Any = "VOID",
	var component: Component? = null,
	var actor: Actor? = null,
	var group: Group? = null,
	var parent: R2DNode? = null,
	var children: MutableList<R2DNode> = ArrayList()
) {

	override fun toString(): String {
		val sb = StringBuilder()
		sb.append("[ ")
		sb.append("\ntype = ${type}")
		sb.append("\ncomponent = ${component}")
		sb.append("\nactor = ${actor?.name}")
		sb.append("\ngroup = ${actor?.name}")
		sb.append("\nparent = ${parent?.type}")
		sb.append("\nchildren = ${children}")
		sb.append(" ]")
		return sb.toString()
	}
}